Apps/phcorelight/corelight_connector.py

```diff
@@ -0,0 +1,573 @@
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_check_results(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        location = param['location']
+        ret_val, response = self._make_rest_call(location, action_result, params=None, headers=None)
+
+        if (phantom.is_fail(ret_val)):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def handle_action(self, param):
+
+        ret_val = phantom.APP_SUCCESS
+
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id == 'test_connectivity':
+            ret_val = self._handle_test_connectivity(param)
+
+        elif action_id == 'ip_intel':
+            ret_val = self._handle_ip_intel(param)
+
+        elif action_id == 'get_conf':
+            ret_val = self._handle_get_conf(param)
+
+        elif action_id == 'list_intel_framework':
+            ret_val = self._handle_list_intel_framework(param)
+
+        elif action_id == 'list_input_framework':
+            ret_val = self._handle_list_input_framework(param)
+
+        elif action_id == 'input_framework':
+            ret_val = self._handle_input_framework(param)
+
+        elif action_id == 'backup_corelight':
+            ret_val = self._handle_backup_corelight(param)
+
+        elif action_id == 'restore_corelight':
+            ret_val = self._handle_restore_corelight(param)
+
+        elif action_id == 'check_results':
+            ret_val = self._handle_check_results(param)
+
+        return ret_val
+
+    def initialize(self):
+
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
+        self._state = self.load_state()
+
+        # get the asset config
+        config = self.get_config()
+
+        """
+        # Access values in asset config by the name
+
+        # Required values can be accessed directly
+        required_config_name = config['required_config_name']
+
+        # Optional values should use the .get() function
+        optional_config_name = config.get('optional_config_name')
+        """
+
+        self._base_url = config.get('base_url')
+        self._user = config.get('user')
+        self._password = config.get('password')
+
+        if self.get_action_identifier() in ['ip_intel', 'list_intel_framework', 'input_framework', 'backup_corelight']:
+            if hasattr(Vault, 'get_vault_tmp_dir'):
+                temp_dir = Vault.get_vault_tmp_dir()
+            else:
+                temp_dir = 'opt/phantom/vault/tmp'
+            guid = uuid.uuid4()
+            self._local_dir = temp_dir + '/{}'.format(guid)
```

Comment:
I don't _love_ that they are using the vault temp directory for many of their actions. But I guess it works?

Reply:
As discussed, not fixing anything in this